sql: run TestLintClusterSettingNames on CCL code

yuzefovich opened the pull request:
This commits enables running `TestLintClusterSettingNames` on CCL code as well as fixes some minor existing failures. I examined all of the cluster settings that are recommended to be named differently, and all of them are present on 23.1, so they have been grandfathered in.

Found while working on #76378.
Epic: None.

Release note: None

yuzefovich committed:
ts: remove mentions of old deprecated settings

This commit removes all mentions of
`timeseries.storage.10s_resolution_ttl` and
`timeseries.storage.30m_resolution_ttl` cluster settings that have been
deprecated long time ago. I believe they were kept around due to the
migration, but the permanent upgrades introduced in
399e56b3f56c0dce6c34525bebf58297be320083 seem to have removed the need
for that.

Release note: None

yuzefovich committed:
sql: run TestLintClusterSettingNames on CCL code

This commits enables running `TestLintClusterSettingNames` on CCL code
as well as fixes some minor existing failures. I examined all of the
cluster settings that are recommended to be named differently, and all
of them are present on 23.1, so they have been grandfathered in.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106947)
<!-- Reviewable:end -->

miretskiy reviewed (approved):
changfeed +1

yuzefovich committed:
sqlnoccltest: extract new test package that doesn't import CCL

This allows for the single test in this package (that expects that CCL
code is not loaded) to pass when `ccl` package is loaded by `sql` tests.
There is currently no enforcement that `sqlnoccltest` doesn't import
`ccl`. This adjustment is needed because the following commit introduces
a dependency on `ccl` into `sql_test`.

The alternative solution would be to adjust a couple of expected error
messages in `TestRemovePartitioningOSS` to match the behavior when `ccl`
is loaded.

Release note: None

yuzefovich commented:
Apologies for all the pushes, RFAL.
